### PR TITLE
feat(cmf): add redux-storage support

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

When we reload the page we lost the state of the app.
We would like to keep part of the state using localstorage
For example if you dock the sidepanel and you reload, the sidepanel is not docked.

**What is the chosen solution to this problem?**

with this plugin you can install the needed deps in your project and configure what you want to keep in sync.
We have used this implementation to sync the store in the localstorage on the beforeunload event.

example of usage:
```

function appFactory(storage = {}) {
	const { initialState, engine } = storage;
	const store = cmfstore.initialize(
		appReducer,
		initialState,
		undefined,
		[],
	);
	store.dispatch(actions.settingsActions.fetchSettings('/settings.json'));
	render(
		<App store={store} history={syncHistoryWithStore(browserHistory, store)} />,
		document.getElementById('app'),
	);
	if (engine) {
		reduxLocalStorage.saveOnReload({ engine, store });
	}
}

reduxLocalStorage.loadInitialState({
	key: 'data-streams-redux',
	whitelist: [
		['cmf', 'components', 'SidePanel'],
		['cmf', 'components', 'Container(Form)'],
	],
}).then((storage) => {
	appFactory(storage);
}, (error) => {
	console.error(error);  // eslint-disable-line no-console
	appFactory();
});
```

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/cmf/src/reduxstorage/reduxLocalStorage.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import createEngine from 'redux-storage-engine-localstorage';
+import filter from 'redux-storage-decorator-filter';
+import immutablejs from './redux-storage-decorator-immutablejs';
+// import { save as actionSave } from './actions';
+const CMF_IMMUTABLE_PATHS = [
+	['cmf', 'components'],
+	['cmf', 'collections'],
+];
+
+const CMF_MIDDLEWARE_BLACK_LIST = [
```
Comment (talend-sromain) [inline packages/cmf/src/reduxstorage/reduxLocalStorage.js]: Maybe we can add more things in blacklist with cmf configuration ?

File: output/cmf.eslint.txt [hunk 2]
```diff
@@ -9,5 +9,14 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/cmf/src/componentState.js
   36:3  warning  Unexpected console statement  no-console
 
-✖ 1 problem (0 errors, 1 warning)
+/home/travis/build/Talend/ui/packages/cmf/src/reduxstorage/redux-storage-decorator-immutablejs.js
```
Comment (talend-sromain) [inline output/cmf.eslint.txt]: Some lint here

File: packages/cmf/src/reduxstorage/reduxLocalStorage.js (hunk 1 above)
Comment (jmfrancois) [inline packages/cmf/src/reduxstorage/reduxLocalStorage.js]: for sure !

File: packages/cmf/src/reduxstorage/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ * This is a CMF plugin that let you configure your store
+ * "redux-storage": "^4.1.2",
+ * "redux-storage-decorator-filter": "^1.1.8",
+ * "redux-storage-decorator-immutablejs": "^1.0.4",
+ * "redux-storage-engine-localstorage": "^1.1.4",
```
Comment (talend-glorieux) [inline packages/cmf/src/reduxstorage/index.js]: Why is there packages version in comments?

File: packages/cmf/src/reduxstorage/reduxLocalStorage.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/no-unresolved */
+
+import * as storage from 'redux-storage';
+import createEngine from 'redux-storage-engine-localstorage';
+import filter from 'redux-storage-decorator-filter';
+import immutablejs from './redux-storage-decorator-immutablejs';
+// import { save as actionSave } from './actions';
```
Comment (talend-glorieux) [inline packages/cmf/src/reduxstorage/reduxLocalStorage.js]: Please remove unused import.

File: packages/cmf/src/reduxstorage/reduxLocalStorage.js (hunk 1 above)
Comment (jsomsanith) [inline packages/cmf/src/reduxstorage/reduxLocalStorage.js]: Let's add that when we have the need

**Pull request merged**